staging/src/k8s.io/component-base/metrics/prometheusextension/weighted_histogram.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheusextension
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// WeightedObserver
+type WeightedObserver interface {
+	// Set the variable to the given value.
+	Observe(value float64, weight uint64)
+}
+
+type WeightedHistogram interface {
+	prometheus.Metric
+	prometheus.Collector
+	WeightedObserver
+}
+
+//
+type WeightedHistogramOpts struct {
+	Namespace   string
```

Comment:
Can we just switch to 
```
type WeightedHistogramOpts prometheus.HistogramOpts
```

?

Reply:
I would not be opposed to that.  Or how about `type WeightedHistogramOpts = prometheus.HistogramOpts` ?

Reply:
In my latest push I used the type equation.